centreon/www/include/configuration/configObject/service_categories/DB-Func.php

```diff
@@ -308,18 +404,52 @@ function updateServiceCategorieInDB()
 
     updateServiceCategoriesServices($scId);
     $centreon->user->access->updateACL();
+    $fields = CentreonLogAction::prepareChanges($formValues);
+    $centreon->CentreonLogAction->insertLog(
+        object_type: ActionLog::OBJECT_TYPE_SERVICECATEGORIES,
+        object_id: $scId,
+        object_name: $scName,
+        action_type: ActionLog::ACTION_TYPE_CHANGE,
+        fields: $fields
+    );
 }
 
-function deleteServiceCategorieInDB($ids = null)
+function deleteServiceCategorieInDB($serviceCategoryIds = null)
 {
     global $pearDB, $centreon;
 
-    foreach (array_keys($ids) as $key) {
-        $scId = filter_var($key, FILTER_VALIDATE_INT);
-        $query = "DELETE FROM `service_categories` WHERE `sc_id` = :sc_id";
-        $statement = $pearDB->prepare($query);
-        $statement->bindValue(':sc_id', $scId, \PDO::PARAM_INT);
-        $statement->execute();
+    if (! $serviceCategoryIds) {
+        return;
+    }
+
+    $deleteStatement = $pearDB->prepare(
+        <<<SQL
+            DELETE FROM `service_categories`
+            WHERE `sc_id` = :sc_id
+        SQL
+    );
+    $selectStatement = $pearDB->prepare(
+        <<<SQL
+            SELECT sc_name FROM `service_categories`
+            WHERE `sc_id` = :serviceCategoryId LIMIT 1
+        SQL
+    );
+    foreach (array_keys($serviceCategoryIds) as $serviceCategoryId) {
+        $serviceCategoryId = filter_var($serviceCategoryId, FILTER_VALIDATE_INT)
+            ?: throw new \Exception("Invalid service category id");
+
+        $selectStatement->bindValue(':serviceCategoryId', $serviceCategoryId, \PDO::PARAM_INT);
+        $selectStatement->execute();
+        $result = $selectStatement->fetch();
+
+        $deleteStatement->bindValue(':sc_id', $serviceCategoryId, \PDO::PARAM_INT);
+        $deleteStatement->execute();
```

Comment:
You can use the new methods of CentreonDb now ;) Don't forget to catch the exception ;)

Reply:
good